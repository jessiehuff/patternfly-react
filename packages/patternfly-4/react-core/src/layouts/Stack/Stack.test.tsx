import * as React from 'react';
import { mount } from 'enzyme';
import { Stack } from './Stack';
import { StackItem } from './StackItem';

test('isMain set to true', () => {
  const view = mount(
    <Stack>
      <StackItem isFilled>Filled content</StackItem>
    </Stack>
  );
  expect(view).toMatchSnapshot();
});

test('isMain defaults to false', () => {
  const view = mount(
    <Stack>
      <StackItem>Basic content</StackItem>
    </Stack>
  );
  expect(view).toMatchSnapshot();
});

[true, false].forEach(hasGutter => {
  test(`Gutter ${hasGutter}`, () => {
    const view = mount(
      <Stack hasGutter={hasGutter}>
        <StackItem>Basic content</StackItem>
      </Stack>
    );
    expect(view).toMatchSnapshot();
  });
});
