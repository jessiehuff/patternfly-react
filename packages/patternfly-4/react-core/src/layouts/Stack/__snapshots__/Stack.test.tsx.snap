// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gutter false 1`] = `
<Component
  hasGutter={false}
>
  <div
    className="pf-l-stack"
  >
    <Component>
      <div
        className="pf-l-stack__item"
      >
        Basic content
      </div>
    </Component>
  </div>
</Component>
`;

exports[`Gutter true 1`] = `
<Component
  hasGutter={true}
>
  <div
    className="pf-l-stack pf-m-gutter"
  >
    <Component>
      <div
        className="pf-l-stack__item"
      >
        Basic content
      </div>
    </Component>
  </div>
</Component>
`;

exports[`isMain defaults to false 1`] = `
<Component>
  <div
    className="pf-l-stack"
  >
    <Component>
      <div
        className="pf-l-stack__item"
      >
        Basic content
      </div>
    </Component>
  </div>
</Component>
`;

exports[`isMain set to true 1`] = `
<Component>
  <div
    className="pf-l-stack"
  >
    <Component
      isFilled={true}
    >
      <div
        className="pf-l-stack__item pf-m-fill"
      >
        Filled content
      </div>
    </Component>
  </div>
</Component>
`;
